packages/set/src/index.ts

```diff
@@ -18,74 +18,84 @@ const $KEYS = Symbol("track-keys");
 export class ReactiveSet<T> extends Set<T> {
   #triggers = new TriggerCache<T | typeof $KEYS>();
 
-  constructor(values?: Iterable<T> | null) {
+  constructor(values?: readonly T[] | null) {
     super();
-    if (values) for (const v of values) super.add(v);
+    if (values) for (const value of values) super.add(value);
+  }
+
+  [Symbol.iterator](): IterableIterator<T> {
+    return this.values();
   }
 
-  // reads
   get size(): number {
     this.#triggers.track($KEYS);
     return super.size;
   }
-  has(v: T): boolean {
-    this.#triggers.track(v);
-    return super.has(v);
+
+  keys(): IterableIterator<T> {
+    return this.values();
   }
 
-  *keys(): IterableIterator<T> {
-    for (const key of super.keys()) {
-      this.#triggers.track(key);
-      yield key;
+  *values(): IterableIterator<T> {
+    for (const value of super.values()) {
+      this.#triggers.track(value);
+      yield value;
     }
     this.#triggers.track($KEYS);
   }
-  values(): IterableIterator<T> {
-    return this.keys();
-  }
+
   *entries(): IterableIterator<[T, T]> {
-    for (const key of super.keys()) {
-      this.#triggers.track(key);
-      yield [key, key];
+    for (const entry of super.entries()) {
+      this.#triggers.track(entry[0]);
+      yield entry;
     }
     this.#triggers.track($KEYS);
   }
 
-  [Symbol.iterator](): IterableIterator<T> {
-    return this.values();
-  }
-  forEach(callbackfn: (value: T, value2: T, set: this) => void) {
+  forEach(fn: (value1: T, value2: T, set: Set<T>) => void): void {
     this.#triggers.track($KEYS);
-    super.forEach(callbackfn as any);
+
+    for (const value of super.values()) {
+      this.#triggers.track(value);
```

Comment:
While using forEach on Set we could track the used values to trigger a recomputation when a value gets updated.